src/world/shipparts/heal.rs

```diff
@@ -0,0 +1,60 @@
+use crate::timer::Timer;
+use crate::world::types::{Location, Module, ModuleInputs, WorldEvent};
+use mlua::prelude::{LuaFunction, LuaTable};
+use mlua::{Lua, RegistryKey, Result, ToLua, UserData, UserDataFields, UserDataMethods};
+
+struct Heal {
+    timer: Timer,
+    hull: RegistryKey,
+}
+
+impl Module for Heal {
+    fn update(&mut self, lua: &Lua, inputs: ModuleInputs, _: Location) -> Option<WorldEvent> {
+        if self.timer.ready(inputs.dt) {
+            let hull: LuaTable = lua.registry_value(&self.hull).unwrap();
+            let user_data: LuaTable = hull.get("userData").unwrap();
+            let repair: LuaFunction = user_data.get("repair").unwrap();
+            repair.call::<i32, ()>(1).unwrap();
```

Comment:
`self.hull` is a reference to the hull module that we need to hold onto between calls into Rust code. All Lua objects have a lifetime of `'lua`, which is bound to the life of the `Lua` object that we receive in all callbacks from Lua into Rust code. That Lua object only lives until the end of the Rust callback, so we can't keep any Lua objects that live longer than a function call.

mlua's Lua registry is a way around that. It's basically a global table where we can store things and ask for them again later. self.hull is a key that we can use to look up the actual hull module in the Lua registry.